plugins/database/hana/hana_test.go

```diff
@@ -22,122 +22,231 @@ func TestHANA_Initialize(t *testing.T) {
+					RoleName:    "test-test",
+				},
+				Statements: dbplugin.Statements{},
+				Password:   "AG4qagho_dsvZ",
+				Expiration: time.Now().Add(1 * time.Second),
+			},
+			expectErr:     true,
+			assertUser:    assertCredsDoNotExist,
+		},
+		"with creation statements": {
+			req: dbplugin.NewUserRequest{
+				UsernameConfig: dbplugin.UsernameMetadata{
+					DisplayName: "test-test",
+					RoleName:    "test-test",
+				},
+				Statements: dbplugin.Statements{
+					Commands: []string{testHANARole},
+				},
+				Password:   "AG4qagho_dsvZ",
+				Expiration: time.Now().Add(1 * time.Second),
+			},
+			expectErr:     false,
+			assertUser:    assertCredsExist,
+		},
 	}
 
-	usernameConfig := dbplugin.UsernameConfig{
-		DisplayName: "test-test",
-		RoleName:    "test-test",
-	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			connectionDetails := map[string]interface{}{
+				"connection_url": connURL,
+			}
 
-	// Test with no configured Creation Statement
-	_, _, err = db.CreateUser(context.Background(), dbplugin.Statements{}, usernameConfig, time.Now().Add(time.Hour))
-	if err == nil {
-		t.Fatal("Expected error when no creation statement is provided")
-	}
+			initReq := dbplugin.InitializeRequest{
+				Config:           connectionDetails,
+				VerifyConnection: true,
+			}
 
-	statements := dbplugin.Statements{
-		Creation: []string{testHANARole},
-	}
+			db := new()
+			dbtesting.AssertInitialize(t, db, initReq)
+			defer dbtesting.AssertClose(t, db)
 
-	username, password, err := db.CreateUser(context.Background(), statements, usernameConfig, time.Now().Add(time.Hour))
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
+			createResp, err := db.NewUser(context.Background(), test.req)
+			if test.expectErr && err == nil {
+				t.Fatalf("err expected, received nil")
+			}
+			if !test.expectErr && err != nil {
+				t.Fatalf("no error expected, got: %s", err)
+			}
 
-	if err = testCredsExist(t, connURL, username, password); err != nil {
-		t.Fatalf("Could not connect with new credentials: %s", err)
+			test.assertUser(t, connURL, createResp.Username, test.req.Password)
+		})
 	}
 }
 
-func TestHANA_RevokeUser(t *testing.T) {
+func TestHANA_UpdateUser(t *testing.T) {
 	if os.Getenv("HANA_URL") == "" || os.Getenv("VAULT_ACC") != "1" {
 		t.SkipNow()
 	}
 	connURL := os.Getenv("HANA_URL")
 
-	connectionDetails := map[string]interface{}{
-		"connection_url": connURL,
+	type testCase struct {
+		req              dbplugin.UpdateUserRequest
+		startingPassword string
+		expectErrOnLogin bool
 	}
 
-	db := new()
-	_, err := db.Init(context.Background(), connectionDetails, true)
-	if err != nil {
-		t.Fatalf("err: %s", err)
+	tests := map[string]testCase{
+		"no update statements": {
+			req: dbplugin.UpdateUserRequest{
+				Password: &dbplugin.ChangePassword{
+					NewPassword: "this_is_ALSO_Thirty_2_characters_",
+				},
+			},
+			startingPassword: "this_is_Thirty_2_characters_wow_",
+			expectErrOnLogin: true,
```

Comment:
I'm curious why this is expected to fail. I thought there was a default statement used if no update statements are included with a password change?

Reply:
It's expected to fail because logging in after modifying the user prompts that user to change their password by default. (I lost the check that I was using to specifically check for that error message when I converted to the multi-test style.

Reply:
Added the ability to check the error message received. We expect an error because the db wants the user to change their password on first log in after being modified.

Reply:
Oh interesting, thanks for the explanation!